### PR TITLE
feat(boundary): Homepage and Use Case pages

### PR DESCRIPTION
**Pull request by alexcarpenter**
## ✅ "Ready for Review" checklist

- [x] The Vercel preview link has been added to this PR's description
- [x] The Asana task has been added to this PR's description
- [x] This PR's link has been added to the "📐 Pull Request" field on the Asana task
- [ ] This Vercel preview link has been added to the "🔍 Preview" field on the Asana task
- [x] The description template has been filled in below

---

## 🔗 Relevant links

- [Preview link](https://dev-portal-git-acboundary-homepage-usecases-hashicorp.vercel.app/) 🔎
- [Asana task](https://app.asana.com/0/1200641486648086/1202347960758182) 🎟️

## 🗒️ What

Implements Boundary homepage and use case pages to follow the latest io site updates. Currently Boundary does not have any use cases for launch so there was extra code to conditionally render use case sections on the homepage and within the nav.

## 💭 Anything else?

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1202347960758182

### CODE REVIEW

File: src/layouts/_proxied-dot-io/boundary/index.tsx [hunk 1]
```diff
@@ -11,24 +10,32 @@ import localConsentManagerServices from 'lib/consent-manager-services/io-sites'
 import Footer from 'components/_proxied-dot-io/boundary/footer'
 import ProductSubnav from 'components/_proxied-dot-io/boundary/subnav'
 import productData from 'data/boundary.json'
+import query from './query.graphql'
+
+interface Props {
+	/**
+	 * Data from data which may contain nav items for the use cases nav
```
Comment (dstaley) [inline src/layouts/_proxied-dot-io/boundary/index.tsx:17]:
```suggestion
	 * Data from Dato which may contain nav items for the use cases nav
```

File: src/layouts/_proxied-dot-io/boundary/index.tsx [hunk 2]
```diff
@@ -49,13 +56,59 @@ function BoundaryIoLayout({
...
+							text: 'Overview',
+							url: '/',
+							type: 'inbound',
+						},
+						'divider',
+						useCaseNavItems.length > 0
+							? {
+									text: 'Use Cases',
+									submenu: [
+										...useCaseNavItems.map((item) => {
```
Comment (dstaley) [inline src/layouts/_proxied-dot-io/boundary/index.tsx:71]: I don't think you need the spread operator here. Instead, it'd be something like:

```ts
submenu: useCaseNavItems.map((item) => {
```

**Review by dstaley — APPROVED** [review]
This looks good to me! I'd double check the images in the `src/pages/_proxied-dot-io/boundary/use-cases/img` folder since most of them look to be Nomad images.

File: src/layouts/_proxied-dot-io/boundary/index.tsx (hunk 2 above)
Comment (alexcarpenter) [reply src/layouts/_proxied-dot-io/boundary/index.tsx]: https://github.com/hashicorp/dev-portal/pull/526/commits/a80d00f526dc25c3f090a01dde16ecca9222e1f8

**Review by dstaley — APPROVED** [review]

**Pull request merged**